cowbird/handlers/impl/filesystem.py

```diff
@@ -1,79 +1,449 @@
+            if not overwrite and access_allowed:
+                # Hardlink already exists, nothing to do.
+                return
+            # Delete the existing file at the destination path to reset the hardlink path with the expected source.
+            LOGGER.warning("Removing existing hardlink destination path at `%s` to generate hardlink for the newly "
+                           "created file.", hardlink_path)
+            os.remove(hardlink_path)
+
+        self.create_hardlink_path(src_path, hardlink_path, access_allowed, is_parent_dir_writable)
+
+    def on_created(self, path: str) -> None:
+        """
+        Call when a new path is found.
+
+        :param path: Absolute path of a new file/directory
+        """
+        if not os.path.isdir(path) and Path(self.wps_outputs_dir) in Path(path).parents:
+            # Only process files, since hardlinks are not permitted on directories
+            LOGGER.info("Creating hardlink for the new file path `%s`", path)
+            self._create_wpsoutputs_hardlink(src_path=path, overwrite=True)
+
+    def on_modified(self, path: str) -> None:
+        """
+        Called when a path is updated.
+
+        :param path: Absolute path of a new file/directory
+        """
+        # Nothing to do for files in the wps_outputs folder.
+        # Permission modifications (e.g.: via `chmod`) are not supported to simplify the management of wpsoutputs
+        # permissions. Any permission modifications should be done via Magpie, which will synchronize the permissions on
+        # any related hardlinks automatically.
+
+    def _delete_wpsoutputs_hardlink(self, src_path: str,
+                                    process_user_paths: bool = True, process_public_paths: bool = True) -> bool:
+        """
+        Deletes the hardlink path that corresponds to the input source path.
+
+        Returns a bool to indicate if a hardlink path was deleted or not.
+        """
+        regex_match = re.search(self.wps_outputs_user_data_regex, src_path)
+        try:
+            if regex_match:  # user paths
+                if not process_user_paths:
+                    return False
+                magpie_handler = HandlerFactory().get_handler("Magpie")
+                user_name = magpie_handler.get_user_name_from_user_id(int(regex_match.group(2)))
+                linked_path = self.get_user_hardlink(src_path=src_path,
+                                                     bird_name=regex_match.group(1),
+                                                     user_name=user_name,
+                                                     subpath=regex_match.group(3))
+            else:  # public paths
+                if not process_public_paths:
+                    return False
+                linked_path = self._get_public_hardlink(src_path)
+            if os.path.isdir(linked_path):
+                os.rmdir(linked_path)
+            else:
+                os.remove(linked_path)
```

Comment:
As mentionned in earlier responses, the destination folders containing the hardlinks are assumed not to contain anything important since it should only contain hardlinks to files from the source WPS outputs dir. Destination folders should not be used to stored anything else, so here we delete the path regardless of the content.